src/user/user.controller.ts

```
@@ -43,16 +44,24 @@ export class UserController {
     if (!req.user || !req.body.email) {
       throw new BadRequestException('Invalid user data.');
     }
-    return this.userService.findByEmail(req.body.email);
+    const user = await this.userService.findByEmail(req.body.email);
```

Move this logic to the service class

If the user is not found, it should throw an exception to return a 404 status code, instead of a custom error message

---

I thought of doing that at first, but then I thought it would make more sense to acknowledge that the user still exists, and that its account has been deleted. In any case, let me know if I should still throw the NotFoundException and I'll do it

---

But currently if the user exists, it has a profile. It it doesn't exist, it doesn't hav a profile. It's a One-to-One kind of relationship

---

fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/27/commits/5657f8ccd61312b517ada70345cd19d2158d162b